articles/choosing-between-firebase-and-mongodb/index.md

```diff
@@ -0,0 +1,95 @@
+5. Indexing.
+MongoDB indexes improve search performance.
+### Advantages of MongoDB
+1. It contains heterogeneous data.
+2. It provides high performance, scalability, and availability.
+3. It can be integrated with Big Data Hadoop.
+4. MongoDB's documentation has an extensive collection of literature and MongoDB tutorials for new users.
+5. MongoDB has high data security since there is no risk of SQL injection.
+### Disadvantages of MongoDB
+1. There is high data consumption because of denormalization.
+2. It has no joins. We need to make many queries and join data.
+3. MongoDB has a limited data size.
+4. Indexing and searching in MongoDB is not powerful.
+### History of Firebase
+Firebase is a cloud service provider and backend as a service company based in San Francisco, California, founded in September 2011 by James Tamplin and Andrew Lee as Envolve. It provided developers with an Application Programming Interface(API) that enabled the integration of online chat functionality into their website. It provided developers with an Application Programming Interface(API) that enabled the integration of online chat functionality into their website.
+### What is firebase
+According to [Cypress](https://www.cypressoft.com/post/choose-firebase-over-backend-development/),Firebase is a toolkit and infrastructure that aims at supporting the process of building better applications and websites.
+Firebase is a Back-end as a service owned by Google which provides a server-less backend to app and web developers. Firebase offers a wide range of backend functionalities which include authentication,real-time database, storage, analytics, crash reporting, and dynamic links.
+### Features of Firebase
+1. Real-Time Database
+2. Hosting
+3. Authentication
+Firebase comes with a built-in authentication module. It supports Facebook, Gmail, Twitter, and password login support.
+4. Firebase Storage
+5. Firebase Analytics
+### Advantages of Firebase
+1. It is free.
+Developers can create as more than one project in Firebase for free. Developers don't need to buy a premium license to use Firebase.
+2. Concise Documentation
+Firebase SDK is documented and has plenty of examples over its website.
+3. Improved Database
+Firebase is an improved database since it provides better performance by storing large chunks of data in JSON format.
+4. It is fast.
+Applications and websites developed using Firebase are faster because its cloud services lead to faster fetching and displaying of data on the screens.
+### Disadvantages of using Firebase
+1. Hosted solution
+Firebase is a hosted solution so the user doesn't have any root access to the location where the data is stored.
+2. Costs
+Firebase free plan limits up to 50 connections and 1,000 Megabytes of storage.
+3. Data Migration Problem.
+Firebase stores its data in JSON format so managing large amounts of structured data need the developer to write powerful queries in SQL.
```

Comment:
JSON format -> just "JSON"

Need commas before clauses i.e. "so"

"need the developer" => replace need with require